src/AccessibilityInsights.Extensions.TelemetryTests/AITelemetryTests.cs

```diff
@@ -1,69 +1,151 @@
+
+            var eventWithoutProperty = testEvents[1];
+            Assert.IsFalse(eventWithoutProperty.Properties.ContainsKey(property), "The event property from the first event must not exist on the second event");
+
+            wrapperMock.VerifyAll();
+        }
+
+        [TestMethod]
+        [Timeout(1000)]
+        public void PublishEvent_ContextPropertiesArePersistent()
+        {
+            List<EventTelemetry> testEvents = new List<EventTelemetry>();
+
+            Mock<ITelemetryClientWrapper> wrapperMock = new Mock<ITelemetryClientWrapper>(MockBehavior.Strict);
+            wrapperMock.Setup(x => x.TrackEvent(It.IsAny<EventTelemetry>()))
+                .Callback<EventTelemetry>((e) => testEvents.Add(e));
+
+            var aiTelemetry = new AITelemetry(wrapperMock.Object);
+
+            const string action1 = "Action 1";
+            const string action2 = "Action 2";
+            const string property = "Context Property";
+            const string value = "First";
+            aiTelemetry.AddOrUpdateContextProperty(property, value);
+            aiTelemetry.PublishEvent(action1);
+            aiTelemetry.PublishEvent(action2);
+
+            for (int loop = 0; loop < 2; loop++)
             {
-                List<EventTelemetry> testEvents = new List<EventTelemetry>();
-                ShimAITelemetry.AllInstances.ProcessEventEventTelemetry = (_, eve) => testEvents.Add(eve);
-                var aiTelemetry = new AITelemetry();
-
-                const string action1 = "Action 1";
-                const string action2 = "Action 2";
-                const string property = "Transient Property";
-                aiTelemetry.PublishEvent(action1, new Dictionary<string, string>
-                {
-                    { property, "SomeValue" }
-                });
-                aiTelemetry.PublishEvent(action2);
-
-                var eventWithProperty = testEvents[0];
-                Assert.IsTrue(eventWithProperty.Properties.ContainsKey(property), "The event property must exist on the first event");
-
-                var eventWithoutProperty = testEvents[1];
-                Assert.IsFalse(eventWithoutProperty.Properties.ContainsKey(property), "The event property from the first event must not exist on the second event");
+                var eventWithContextProperty = testEvents.Find(e => e.Properties.ContainsKey(property));
+                Assert.IsNotNull(eventWithContextProperty);
+                Assert.IsTrue(eventWithContextProperty.Properties[property].Equals(value));
             }
+
+            wrapperMock.VerifyAll();
         }
 
         [TestMethod]
         [Timeout(1000)]
-        public void ContextPropertiesRemainBetweenEventsTest()
+        public void ReportException_ExceptionIsNull_DoesNotCallWrapper()
         {
-            using (ShimsContext.Create())
+            Mock<ITelemetryClientWrapper> wrapperMock = new Mock<ITelemetryClientWrapper>(MockBehavior.Strict);
```

Comment:
Does moq let us verify a non-call more explicitly?

Reply:
With `MockBehavior.Strict`, the mock will throw an Exception on any attempt to call a method that doesn't have a corresponding `Setup` call.